apps/src/sites/studio/pages/components/editorRedux.js

```diff
@@ -0,0 +1,127 @@
+/* global levelKeyList */
+
+import _ from 'lodash';
+
+function updatePositions(node) {
+  for (var i = 0; i < node.length; i++) {
+    node[i].position = i + 1;
+  }
+}
+
+let newStageId = -1;
+let newLevelId = -1;
+
+export default function reducer(state, action) {
```

Comment:
I'm surprised you don't get a warning about needing an initial state. I thought redux usually complained if calling `reducer(undefined, action)` returns undefined. I would make this `reducer(state=[], action) {`